activesupport/lib/active_support/messages/rotator.rb

```diff
@@ -47,7 +47,10 @@ def build_rotation(secret = @secret, options)
         def run_rotations(on_rotation)
           @rotations.find do |rotation|
             if message = yield(rotation) rescue next
-              on_rotation&.call
+              ActiveSupport::Notifications.instrument('rotation.active_support') do |payload|
+                on_rotation&.call(payload)
```

Comment:
Pass the payload here, in order to be able to add information, notably the name of the cookie on which there was a rotation. This really helped us to have a better view of the propagation.

I haven't found a better way to do it. And it results in a breaking change: https://github.com/rails/rails/blob/4cc564f156b21fcbdcb518182adf557c45e58e40/activesupport/test/secure_compare_rotator_test.rb#L41

I'll let you tell me what you think.